Traceback (most recent call last):
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/jupyter_cache/executors/utils.py", line 56, in single_nb_execution
    record_timing=False,
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/asyncio/base_events.py", line 583, in run_until_complete
    return future.result()
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/client.py", line 554, in async_execute
    cell, index, execution_count=self.code_cells_executed + 1
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/alandmesser/anaconda3/envs/workshops/lib/python3.7/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
````{Dropdown} L√∂sungen kompakt
# Fehlende Werte ermitteln
missingValuesCount = data.isnull().sum()
print('Fehlede Werte pr√ºfen vor Aufbereitung')
print(missingValuesCount)

# Fehlende Werte ersetzen
data['Gender'].fillna(data['Gender'].mode()[0], inplace=True)
data['Cars'].fillna(data['Cars'].mode()[0], inplace=True)
data['Age'].fillna(data['Age'].median(), inplace=True)
data['Children'].fillna(data['Children'].mode()[0], inplace=True)
data['Marital Status'].fillna(data['Marital Status'].mode()[0], inplace=True)
data['Income'].fillna(data['Income'].mean(), inplace=True)
data['Home Owner'].fillna(data['Home Owner'].mode()[0], inplace=True)

# Fehlende Werte ermitteln/pr√ºfen
missingValuesCount = data.isnull().sum()
print('Fehlede Werte pr√ºfen nach Aufbereitung')
print(missingValuesCount)

# Datensets erstellen
X = data.drop(['Purchased Bike'], axis=1)
y = data['Purchased Bike']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1, stratify=y)
X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, test_size=0.25, random_state=1, stratify=y_train)

# Datensets speichern
datasets = {
    'X_train': X_train,
    'y_train': y_train,
    'X_val': X_val,
    'y_val': y_val,
    'X_test': X_test,
    'y_test': y_test
}

with open('../output/bikebuyers/datasets.pkl', 'wb') as handle:
    pickle.dump(datasets, handle)
    
with open('../output/bikebuyers/datasets.pkl', 'wb') as handle:
    ds = pickle.load(datasets, handle)
    print('Erste Zeilen des geladenen Trainingsdatenset:')
    print(ds['X_train'].head())
------------------

[0;36m  File [0;32m"<ipython-input-22-0e9ed6be8f4e>"[0;36m, line [0;32m1[0m
[0;31m    ````{Dropdown} L√∂sungen kompakt[0m
[0m    ^[0m
[0;31mSyntaxError[0m[0;31m:[0m invalid syntax

SyntaxError: invalid syntax (<ipython-input-22-0e9ed6be8f4e>, line 1)

